node/metrics/src/lib.rs

```diff
@@ -38,9 +40,12 @@ use std::{
 use time::OffsetDateTime;
 
 /// Initializes the metrics and returns a handle to the task running the metrics exporter.
-pub fn initialize_metrics() {
+pub fn initialize_metrics(ip: Option<SocketAddr>) {
     // Build the Prometheus exporter.
-    metrics_exporter_prometheus::PrometheusBuilder::new().install().expect("can't build the prometheus exporter");
+    metrics_exporter_prometheus::PrometheusBuilder::new()
+        .with_http_listener(ip.unwrap_or(SocketAddr::from_str("0.0.0.0:9000").unwrap()))
```

Comment:
I would prefer not to hard-code the default socketAddr like this - `SocketAddr::from_str("0.0.0.0:9000)`.

Even if it's functionally the same as calling `metrics_exporter_prometheus::PrometheusBuilder::new()` now, if the defaults in PrometheusBuilder change in the future, we would need to manually perform the update.

Ideally, we keep the old syntax if `ip` is not specified here. Maybe with a match case.

Reply:
Makes perfect sense, will update after my work shift